torchvision/transforms/autoaugment.py

```diff
@@ -333,7 +333,7 @@ class TrivialAugmentWide(torch.nn.Module):
     r"""Dataset-independent data-augmentation with TrivialAugment Wide, as described in
     `"TrivialAugment: Tuning-free Yet State-of-the-Art Data Augmentation" <https://arxiv.org/abs/2103.10158>`.
     If the image is torch Tensor, it should be of type torch.uint8, and it is expected
-    to have [..., 1 or 3, H, W] shape, where ... means an arbitrary number of leading dimensions.
+    to have [..., 3, H, W] shape, where ... means an arbitrary number of leading dimensions.
     If img is PIL Image, it is expected to be in mode "L" or "RGB".
```

Comment:
```suggestion
    If img is PIL Image, it is expected to be in mode "RGB".
```

Reply:
Hi @datumbox ,

The below snippet still works when I pass a pillow image with the mode is `'L'`. This definition of "work" here is that it does not raise the above issue. I've also checked some of the results and see that they are all good. If then, does it come from the tensor transformation side?

```python
from PIL import Image
from pathlib import Path
import torch
import torchvision
from torchvision import transforms

image_path = 'n02100877_4699.jpg'
raw_image = Image.open(image_path).convert('L')  # try with 'L', 'RGB
print(image.size, image.mode) 
tf = transforms.AutoAugment(transforms.autoaugment.AutoAugmentPolicy.CIFAR10)
for seed in range(0, 2000):
    torch.manual_seed(seed)
    image = tf(raw_image)
    image.save(out_dir / f"{seed}.png", 'png')
```

Reply:
Here is the already setup colab notebook that you could manually check yourself: https://colab.research.google.com/drive/1WtlOAASAGJA927oeFECN2YSFAhlMJ6j2?usp=sharing

Reply:
Hmm, thanks for investigating. It seems that PIL supports the colour operators even when the image is grayscale. 

I think we have a few options:
1. Update the documentation for Tensors only since PIL supports `L`
2. Update the documentation for both to avoid confusing people
3. Update the implementations of colour transforms of TorchVision that don't support Grayscale images to align them with PIL.

I think option 3 is worth exploring but it will take longer and needs to happen carefully. So I would be in favour of doing 1 or 2 in this PR and follow up on 3 on a separate one. I don't have strong opinion over how we go, so let me know what you prefer.

Reply:
Yeah, I think that this PR is just for updating the documentation only to keep it relates to the original issue. Furthermore, since we have already declared some policies such as ImageNet, CIFAR10 that their input is not a grayscale image. Thus, changing to support grayscale images would have some conflicts with the policy definitions as well that we might have further discussion in another thread.

In conclusion, in this PR, I would change the document so that this transformation only supports RGB for now.